cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/WorkflowDataset.java

```diff
@@ -82,6 +82,23 @@ void write(Id.Workflow id, RunRecordMeta runRecordMeta, List<ProgramRun> program
     table.put(rowKey, NODES, Bytes.toBytes(value));
   }
 
+  public void delete(Id.Application id) {
+    Scanner scanner = table.scan(null, null);
```

Comment:
can we avoid scanning the whole table? can we not construct the row key based upon the app id?